src/Files.App/ViewModels/ItemViewModel.cs

```diff
@@ -1878,7 +1878,7 @@ private void WatchForDirectoryChanges(string path, CloudDriveSyncStatus syncStat
 
             if (aProcessQueueAction == null) // Only start one ProcessOperationQueue
             {
-                aProcessQueueAction = Task.Run(() => ProcessOperationQueue(watcherCTS.Token, hasSyncStatus));
+                aProcessQueueAction = Win32API.StartSTATask(() => ProcessOperationQueue(watcherCTS.Token, hasSyncStatus));
```

Comment:
Not a problem but my understanding is that a dedicated thread makes sense only if you have blocking & non cpu-bound work (most of the work here is short running or asynchronous).

If instead there's a need to specify the task is long running perhaps I'd use Task.Factory.StartNew + LongRunning option, I'd reserve StarSTATask for operations that really require a STA thread + OLE (context menu, shell thumbnails..)

Reply:
Done.

Reply:
Thanks, unfortunately Task.factory.StartNew has some bad default parameters, I believe the correct call to be smth like (note expecially the TaskScheduler):

`Task.Factory.StartNew(someAction, CancellationToken.None, TaskCreationOptions.DenyChildAttach | TaskCreationOptions.LongRunning, TaskScheduler.Default)`

Reply:
Note: my knowledge about this topic is a bit poor, so feel free to disregard these suggestions / wskt for other oponions ;)

Reply:
I just checked the document and did't see why we need `TaskCreationOptions.DenyChildAttach`.

Reply:
It would have been only to make it equivalent to Task.Run, but not necessary.

Reply:
PS the important point was TaskScheduler.Default btw. Not passing it risks running the Task on the ui thread / causing a deadlock.

Reply:
Will fix this in following PR.